services/core-api/app/api/incidents/models/mine_incident.py

```diff
@@ -74,33 +75,33 @@ class MineIncident(AuditMixin, Base):
 
     determination_type = db.relationship(
         'MineIncidentDeterminationType',
-        backref='mine_incidents',
+        backref='mine_incident',
         lazy='joined',
         uselist=False,
         foreign_keys=[determination_type_code])
     dangerous_occurrence_subparagraphs = db.relationship(
         'ComplianceArticle',
-        backref='mine_incidents',
+        backref='mine_incident',
         lazy='joined',
         secondary='mine_incident_do_subparagraph')
     followup_investigation_type = db.relationship(
         'MineIncidentFollowupInvestigationType',
-        backref='mine_incidents',
+        backref='mine_incident',
         lazy='joined',
         uselist=False)
-    mine_incident_category_code = db.Column(
-        db.String, db.ForeignKey('mine_incident_category.mine_incident_category_code'))
-    mine_incident_category = db.relationship('MineIncidentCategory', lazy='joined')
 
     recommendations = db.relationship(
         'MineIncidentRecommendation',
         primaryjoin=
         "and_(MineIncidentRecommendation.mine_incident_id == MineIncident.mine_incident_id, MineIncidentRecommendation.deleted_ind==False)",
-        lazy='selectin')
+        lazy='select')
     documents = db.relationship('MineIncidentDocumentXref', lazy='joined')
     mine_documents = db.relationship(
         'MineDocument', lazy='joined', secondary='mine_incident_document_xref')
 
+    mine_incident_category_xref = db.relationship(
+        'MineIncidentCategoryXref', backref='mine_incident', lazy='select')
+
```

Comment:
see above with the use of the xref table as a secondary, can this do the same. 

```mine_incident_categories = db.relationship('MineIncidentCategory', secondary='mine_incident_category_xref', backref='mine_incidents', lazy='select')```

Reply:
Like this?
`    mine_incident_category_xref = db.relationship('MineIncidentCategoryXref', lazy='joined')
    mine_incident_categories = db.relationship(
        'MineIncidentCategory', secondary='mine_incident_category_xref', backref='mine_incidents', lazy='select')`

Would this change actually affect anything (as everything is working now) and require more changes, or is it just to merely better represent the relationships/model?

Reply:
you could leave both, but you've had to write alot of code further down that doesn't match existing patterns, so i would like to see this cleaned up before we merge.

Reply:
Ok. I was looking at the "recommendations" code as a reference for patterns actually, but now I see that looking at the "documents xref" table is a better candidate, as the relationship it has to mine_incident is the same as the new "category xref" table. (?)